File: searchin algorithms/binary_search.py
# Binary search algorithm implementation in python,
# it divides the search space into two equal halves

def Binary_search(arr,target):
    arr.sort()
    left , right = 0 , len(arr)-1

    while left <= right:
        mid = (left + right)//2

        if arr[mid] == target:
            return mid
        elif arr[mid]>target:
            right = mid - 1
        else:
            left = mid + 1  
    return -1

sorted_list = [1, 3, 5, 7, 9, 11, 13, 15]
target_value = 13

result = Binary_search(arr=sorted_list,target=target_value)
if result != -1:
    print(f'founded target element on this position {result} and element is {sorted_list[result]}')
else:
    print(f'element not found !')

File: searchin algorithms/binary_search.cpp
#include <iostream>
using namespace std;
// #include <vector>

int binarySearch(int arr[], int size , int target) {
    int left = 0;
    int right = size - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        // Check if target is present at mid
        if (arr[mid] == target) {
            return mid;
        }

        // If target is greater, ignore left half
        if (arr[mid] > target) {
            right = mid - 1;
        }
        // If target is smaller, ignore right half
        else {
            left = mid + 1;
        }
    }

    // Target was not found in the array
    return -1;
}

int main() {
    int arr[] = {2, 3, 4, 10, 40};
    int target = 30;
    int size = sizeof(arr)/sizeof(arr[0]);

    int result = binarySearch(arr, size, target);

    if (result != -1) {
        cout << "Element found at index " << result << std::endl;
        cout<<"Your item is : "<<arr[result]<<endl;
    } else {
        cout << "Element not found in the array" << std::endl;
    }

    return 0;
}
